core/src/main/java/org/apache/druid/data/input/impl/SqlFirehose.java

```diff
@@ -35,18 +34,18 @@
 public class SqlFirehose implements Firehose
 {
   private final Iterator<JsonIterator<Map<String, Object>>> resultIterator;
-  private final InputRowParser parser;
+  private final MapInputRowParser parser;
   private final Closeable closer;
   private JsonIterator<Map<String, Object>> lineIterator = null;
 
   public SqlFirehose(
-      Iterator lineIterators,
-      InputRowParser<Map<String, Object>> parser,
+      Iterator<JsonIterator<Map<String, Object>>> lineIterators,
+      InputRowParser parser,
       Closeable closer
   )
   {
     this.resultIterator = lineIterators;
-    this.parser = parser;
+    this.parser = new MapInputRowParser(parser.getParseSpec());
```

Comment:
I think you'll need to extract the TransformSpec from the parser (using `TransformSpec.fromInputRowParser`) for transforms to work properly. Could you please look into this and add a test if there isn't one?

Reply:
Added transform handling and tests. This involved moving `SqlFirehose` and some of it's friends out of `druid-core` and into `druid-server` and making some stuff `public`.